drivers/i2c/i2c_sc18is600.c

```diff
@@ -0,0 +1,315 @@
+/*
+ * Copyright (c) 2017-2018 Antmicro Ltd <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <misc/printk.h>
+#include <i2c.h>
+#include <spi.h>
+
+#include "i2c_sc18is600.h"
+
+#define I2C_CS	0
+
+#define I2C_CLOCK_REG	2
+#define I2C_TO_REG	3
+#define I2C_STATUS_REG	4
+#define I2C_ADDR_REG 5
+
+#define I2C_STATUS_SUCCESS		0xF0
+#define I2C_STATUS_NO_ACK		0xF1
+#define I2C_STATUS_NACK			0xF2
+#define I2C_STATUS_BUSY			0xF3
+#define I2C_STATUS_TIMEOUT		0xF8
+#define I2C_STATUS_INVALID_COUNT	0xF9
+
+#define WRITE_N_BYTES_TO_DEVICE		0x00
+#define READ_N_BYTES_FROM_DEVICE	0x01
+#define READ_AFTER_WRITE_DEVICE		0x02
+
+#define READ_BRIDGE_RX_BUFFER		0x06
+#define WRITE_TO_BRIDGE_REGISTER	0x20
+#define READ_FROM_BRIDGE_REGISTER	0x21
+
+struct i2c_sc18is600_runtime {
+	struct device *spi_d;
+	struct spi_config spi_c;
+	u32_t i2c_config;
+};
+
+struct i2c_sc18is600_config {};
+
+static struct i2c_sc18is600_runtime i2c_sc18is600_0_runtime;
+static struct i2c_sc18is600_config i2c_sc18is600_0_config;
+
+
+int i2c_sc18is600_spi_send_buffer(struct device *dev, u8_t *data, size_t len)
+{
+	struct i2c_sc18is600_runtime *run_data =
+		(struct i2c_sc18is600_runtime *) dev->driver_data;
+
+	struct spi_buf tx_buf = {
+		.buf = data,
+		.len = len,
+	};
+
+	int err;
+
+	err = spi_write(&(run_data->spi_c), &tx_buf, 1);
+	if (err) {
+		printk("spi_write failed\n");
```

Comment:
Don't use printk. Use SYS_LOG primitives (see also SYS_LOG_I2C_LEVEL in i2c/Kconfig)